### PR TITLE
Feat(eos_cli_config_gen): Added support for `neighbor x.x.x.x encapsulation mpls next-hop-self source-intf <source-interface>`

### LINKED ISSUE #4243: Add support for "neighbor x.x.x.x encapsulation mpls next-hop-self source-intf <source-interface>"
### Enhancement summary

Just looking to support the following command:

```
address-family evpn
      neighbor x.x.x.x encapsulation mpls next-hop-self source-intf <source-interface>
```

TOI:
https://www.arista.com/en/support/toi/eos-4-25-0f/14591-evpn-l3-gateway

This command is required when configuring EVPN-VXLAN to EVPN-MPLS L3 gateway.

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

please detail your use case

### Describe the solution you would like

This command is required when configuring EVPN-VXLAN to EVPN-MPLS L3 gateway.

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Added support for `neighbor x.x.x.x encapsulation mpls next-hop-self source-intf <source-interface>`

## Related Issue(s)

Fixes #4243 

## Component(s) name

`arista.avd.eos_cli_config_gen

## Proposed changes
Added support for `neighbor x.x.x.x encapsulation mpls next-hop-self source-intf <source-interface>`

## How to test
Run molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by laxmikantchintakindi — APPROVED** [review]

**Review by Shivani-gslab — APPROVED** [review]
LGTM!

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2 [hunk 1]
```diff
@@ -924,8 +924,14 @@ router bgp {{ router_bgp.as }}
...
 {%             endif %}
-{%             if neighbor.encapsulation is arista.avd.defined %}
+{%             if neighbor.encapsulation is arista.avd.defined and neighbor.encapsulation in ["vxlan", "path-selection"] %}
       neighbor {{ neighbor.ip_address }} encapsulation {{ neighbor.encapsulation }}
+{%             elif neighbor.encapsulation is arista.avd.defined("mpls") %}
+{%                 set encapsulation_cli = "neighbor " ~  neighbor.ip_address ~ " encapsulation mpls" %}
+{%                 if neighbor.next_hop_self_source_interface is arista.avd.defined %}
+{%                     set encapsulation_cli = encapsulation_cli ~ " next-hop-self source-interface " ~ neighbor.next_hop_self_source_interface %}
+{%                 endif %}
+      {{ encapsulation_cli }}
```
Comment (Vibhu-gslab) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2]: May be we can write it like:
```
{%             if neighbor.encapsulation is arista.avd.defined  %}
{%                 set encapsulation_cli =  "neighbor " ~ neighbor.ip_address ~ " encapsulation " neighbor.encapsulation %}
{%                 if neighbor.encapsulation == "mpls" and neighbor.next_hop_self_source_interface is arista.avd.defined %}
{%                     set encapsulation_cli = encapsulation_cli ~ " next-hop-self source-interface " ~ neighbor.next_hop_self_source_interface %}
{%                 endif %}
      {{ encapsulation_cli }}
```

Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2]: updated

**Review by ClausHolbechArista — DISMISSED** [review]
Blocking merge until 5.0 has been released.

**Review by Vibhu-gslab — APPROVED** [review]

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/router_bgp.schema.yml [hunk 2]
```diff
@@ -1329,6 +1329,9 @@ keys:
                     - "mpls"
                     - "path-selection"
                   description: Transport encapsulation for the neighbor.
+                next_hop_self_source_interface:
+                  type: str
+                  description: Source interface name for MPLS encapsulation. Requires `encapsulation` to be set as `mpls`.
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/router_bgp.schema.yml:1334]: Can you add the same comment line 1261 for the same key under neighbor_default for consistency? (I cannot comment up there in Github)

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/router_bgp.schema.yml (hunk 2 above)
Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/router_bgp.schema.yml:1334]: added

**Review by gmuloc — APPROVED** [review]

**Pull request merged**